### ISSUE TITLE
conda install git hangs

### ISSUE THREAD

**Issue by msarahan** [conda/conda#6179]
from @amueller via gitter:

this might be more an anaconda issue or a conda issue or both... if I try to "conda install git" the dependency solver hangs
in one instance it suggested downgrading from python3.6 to python2.7
(this is on windows with current anaconda)

**Comment by amueller** [comment]
Thanks for opening the issue.
I had opened an issue here: https://github.com/ContinuumIO/anaconda-issues/issues/6687
as I'm not sure which tracker is more applicable.

**Comment by msarahan** [comment]
Thanks.  Generally, I think conda hanging is always going to be a conda issue, unless there's some bug with the packages that are causing the hang.  I'd expect more widespread issue reports if the latter were the case.

**Comment by amueller** [comment]
True. Could be a combination of both. I'm not sure if this package is used a lot (or at all), so I'm not sure there would be more reports.
It might be for some strange reason the MIPS problem is very hard, I guess you could give it a time limit and then bail? not sure if that's better, though.

**Comment by kalefranz** [comment]
Hey @amueller, not sure what version of conda you're running.  Mind including `conda info`?

Also, if it's a recent version, `conda install git -vvv` will help with understanding some of what the solver is doing.  That output can be pretty verbose, but also pretty helpful.

**Comment by amueller** [comment]
@kalefranz will do when I get to booting windows again. It's whatever's packaged in Anaconda 5.0.0

**Comment by kalefranz** [comment]
Ok that's 4.3.27 I think.  We're recognizing that we're bumping into our limits with the current SAT implementation.  We've already started to experiment with alternatives. https://github.com/conda/conda/pull/6174

**Comment by msarahan** [comment]
I realized last night that git should be a trivially easy case here for the solver, but a brutally hard use of any file verification checks that we do.  The git package on windows is just a repack of git for windows, https://git-scm.com/download/win

It should have zero dependencies.  Indeed:

```
"git-2.11.1-0.tar.bz2": {
      "build": "0",
      "build_number": 0,
      "date": "2017-02-19",
      "depends": [],
      "license": "GPL v2 and LGPL 2.1",
      "license_family": "GPL2",
      "md5": "25f90464d8ca55e85709190dfb4666ea",
      "name": "git",
      "size": 72648165,
      "version": "2.11.1"
    },
    "git-2.6.4-0.tar.bz2": {
      "build": "0",
      "build_number": 0,
      "date": "2016-01-13",
      "depends": [],
      "license": "GPL v2 and LGPL 2.1",
      "license_family": "GPL2",
      "md5": "cd9e9ea7310282658677cb83cbe66dc8",
      "name": "git",
      "size": 64176201,
      "version": "2.6.4"
    },
    "git-2.9.3-0.tar.bz2": {
      "build": "0",
      "build_number": 0,
      "date": "2016-08-25",
      "depends": [],
      "license": "GPL v2 and LGPL 2.1",
      "license_family": "GPL2",
      "md5": "8e86ed9f7d9c6e54a2b856b394d60a61",
      "name": "git",
      "size": 68900780,
      "version": "2.9.3"
    },
```

However, it has tons of small files.  That's a real torture test on the verification stuff in recent conda versions.  We should be able to test this by using Kale's suggestion of -vvv, or possibly also by using --dry-run.  If it's a solver issue, you'll never see the confirmation dialog for carrying out the operation (side question: do you have --always-yes enabled?)

**Comment by mcg1969** [comment]
On the Windows 10 VM on my Mac, I did a fresh install of Anaconda 5 (Python 3.6). `conda install git --dry-run` takes 30 seconds to run, but it offers a plan _that downgrades to Python 2.7_. More as it develops.

**Comment by kalefranz** [comment]
Oh interesting....

**Comment by mcg1969** [comment]
When I update to conda 4.3.30 and try again, it offers a plan that converts everything to vc9. Many of the packages are downgraded from the `main` channel to the `free` channel.

**Comment by mcg1969** [comment]
conda 4.3.27, `conda installl git --dry-run`:
```
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment C:\Users\mcg\Anaconda3:

The following NEW packages will be INSTALLED:

    backports.functools_lru_cache:      1.4-py27h9586e20_1
    backports_abc:                      0.5-py27h0ec6b72_0
    cdecimal:                           2.3-py27h5b739fe_1
    configparser:                       3.5.0-py27h2fa79a8_0
    enum34:                             1.1.6-py27h2aa175b_1
    funcsigs:                           1.0.2-py27h8885ae1_0
    functools32:                        3.2.3.2-py27h0cdbcdb_1
    futures:                            3.1.1-py27h71313b4_0
    git:                                2.14.1-h8288424_2
    grin:                               1.2.1-py27hb905a94_1
    ipaddress:                          1.0.18-py27h08e1fa7_0
    scandir:                            1.5-py27hc48eb83_1
    ssl_match_hostname:                 3.5.0.1-py27hea8a0f4_2
    subprocess32:                       3.2.7-py27hcc576e2_0
    vs2008_runtime:                     9.00.30729.1-hfaea7d5_0

The following packages will be UPDATED:

    _ipyw_jlab_nb_ext_conf:             0.1.0-py36he6757f0_0       --> 0.1.0-py27h9111531_0
    alabaster:                          0.7.10-py36hcd07829_0      --> 0.7.10-py27h2cab13d_0
    anaconda:                           5.0.0-py36hea9b2fc_0       --> 5.0.0-py27hfd113d2_0
    anaconda-client:                    1.6.5-py36hd36550c_0       --> 1.6.5-py27h9642776_0
    anaconda-navigator:                 1.6.8-py36h4b7dd57_0       --> 1.6.8-py27hafd345f_0
    anaconda-project:                   0.8.0-py36h8b3bf89_0       --> 0.8.0-py27h56b7296_0
    asn1crypto:                         0.22.0-py36h8e79faa_1      --> 0.22.0-py27hc975345_1
    astroid:                            1.5.3-py36h9d85297_0       --> 1.5.3-py27h658102e_0
    astropy:                            2.0.2-py36h06391c4_4       --> 2.0.2-py27hbc6ace0_4
    babel:                              2.5.0-py36h35444c1_0       --> 2.5.0-py27h50e9d34_0
    backports:                          1.0-py36h81696a8_1         --> 1.0-py27h6492d98_1
    backports.shutil_get_terminal_size: 1.0.0-py36h79ab834_2       --> 1.0.0-py27h1657125_2
    beautifulsoup4:                     4.6.0-py36hd4cc5e8_1       --> 4.6.0-py27hc287451_1
    bitarray:                           0.8.1-py36h6af124b_0       --> 0.8.1-py27hc9ed29b_0
    bkcharts:                           0.2-py36h7e685f7_0         --> 0.2-py27h92b6de3_0
    blaze:                              0.11.3-py36h8a29ca5_0      --> 0.11.3-py27h97e5449_0
    bleach:                             2.0.0-py36h0a7e3d6_0       --> 2.0.0-py27ha102ec9_0
    bokeh:                              0.12.7-py36h012f572_1      --> 0.12.7-py27h83180d4_1
    boto:                               2.48.0-py36h1a776d2_1      --> 2.48.0-py27h1ccb131_1
    bottleneck:                         1.2.1-py36hd119dfa_0       --> 1.2.1-py27hde90680_0
    bzip2:                              1.0.6-vc14hdec8e7a_1       [vc14] --> 1.0.6-vc9h563fc2e_1        [vc9]
    cachecontrol:                       0.12.3-py36hfe50d7b_0      --> 0.12.3-py27h15eca11_0
    certifi:                            2017.7.27.1-py36h043bc9e_0 --> 2017.7.27.1-py27hf761292_0
    cffi:                               1.10.0-py36hae3d1b5_1      --> 1.10.0-py27hb6f203f_1
    chardet:                            3.0.4-py36h420ce6e_1       --> 3.0.4-py27h56c3b73_1
    click:                              6.7-py36hec8c647_0         --> 6.7-py27hb6defca_0
    cloudpickle:                        0.4.0-py36h639d8dc_0       --> 0.4.0-py27h6dd55d0_0
    clyent:                             1.2.2-py36hb10d595_1       --> 1.2.2-py27h4424948_1
    colorama:                           0.3.9-py36h029ae33_0       --> 0.3.9-py27hdfe4ae1_0
    comtypes:                           1.1.2-py36heb9b3d1_0       --> 1.1.2-py27ha42e334_0
    conda:                              4.3.27-py36hcbae3bd_0      --> 4.3.30-py27he817781_0
    conda-build:                        3.0.22-py36hd71664c_0      --> 3.0.27-py27hdc961ca_0
    conda-verify:                       2.0.0-py36h065de53_0       --> 2.0.0-py27h5217224_0
    contextlib2:                        0.5.5-py36he5d52c0_0       --> 0.5.5-py27h42efda5_0
    cryptography:                       2.0.3-py36h123decb_1       --> 2.0.3-py27h1cc6ae9_1
    curl:                               7.55.1-vc14hdaba4a4_3      [vc14] --> 7.55.1-vc9h3bcf4df_3       [vc9]
    cycler:                             0.10.0-py36h009560c_0      --> 0.10.0-py27h59acbbf_0
    cython:                             0.26.1-py36h18049ac_0      --> 0.26.1-py27h45dccc8_0
    cytoolz:                            0.8.2-py36h547e66e_0       --> 0.8.2-py27h44eb5fa_0
    dask:                               0.15.2-py36h25e1b01_0      --> 0.15.2-py27h293bcc8_0
    dask-core:                          0.15.2-py36hf9e56b0_0      --> 0.15.2-py27h0b47d0a_0
    datashape:                          0.5.4-py36h5770b85_0       --> 0.5.4-py27h3d6e61b_0
    decorator:                          4.1.2-py36he63a57b_0       --> 4.1.2-py27h383e199_0
    distlib:                            0.2.5-py36h51371be_0       --> 0.2.5-py27hbc39c7c_0
    distributed:                        1.18.3-py36h5be4c3e_0      --> 1.18.3-py27h6bb91c2_0
    docutils:                           0.14-py36h6012d8f_0        --> 0.14-py27h8652d09_0
    entrypoints:                        0.2.3-py36hfd66bb0_2       --> 0.2.3-py27h0271f2b_2
    et_xmlfile:                         1.0.1-py36h3d2d736_0       --> 1.0.1-py27h1de5d23_0
    fastcache:                          1.0.2-py36hffdae1b_0       --> 1.0.2-py27hf477a8b_0
    filelock:                           2.0.12-py36hd7ddd41_0      --> 2.0.12-py27h7a543d8_0
    flask:                              0.12.2-py36h98b5e8f_0      --> 0.12.2-py27h30d9212_0
    flask-cors:                         3.0.3-py36h8a3855d_0       --> 3.0.3-py27h4926b05_0
    freetype:                           2.8-vc14h17c9bdf_0         [vc14] --> 2.8-vc9hf582001_0          [vc9]
    gevent:                             1.2.2-py36h342a76c_0       --> 1.2.2-py27h1842022_0
    glob2:                              0.5-py36h11cc1bd_1         --> 0.5-py27h5738e64_1
    greenlet:                           0.4.12-py36ha00ad21_0      --> 0.4.12-py27h32400d3_0
    h5py:                               2.7.0-py36hfbe0a52_1       --> 2.7.0-py27hae8a57c_1
    hdf5:                               1.10.1-vc14hb361328_0      [vc14] --> 1.10.1-vc9h9d13b45_0       [vc9]
    heapdict:                           1.0.0-py36h21fa5f4_0       --> 1.0.0-py27h5528e5e_0
    html5lib:                           0.999999999-py36ha09b1f3_0 --> 0.999999999-py27h47766b3_0
    icu:                                58.2-vc14hc45fdbb_0        [vc14] --> 58.2-vc9h3e79e4e_0         [vc9]
    idna:                               2.6-py36h148d497_1         --> 2.6-py27h1ea29d3_1
    imageio:                            2.2.0-py36had6c2d2_0       --> 2.2.0-py27h283db88_0
    imagesize:                          0.7.1-py36he29f638_0       --> 0.7.1-py27h1482bd8_0
    ipykernel:                          4.6.1-py36hbb77b34_0       --> 4.6.1-py27h30cef79_0
    ipython_genutils:                   0.2.0-py36h3c5d0ee_0       --> 0.2.0-py27hbe997df_0
    ipywidgets:                         7.0.0-py36h2e74ada_0       --> 7.0.0-py27he0f823d_0
    isort:                              4.2.15-py36h6198cc5_0      --> 4.2.15-py27hdc949c3_0
    itsdangerous:                       0.24-py36hb6c5a24_1        --> 0.24-py27hcf63135_1
    jdcal:                              1.3-py36h64a5255_0         --> 1.3-py27h8c72977_0
    jedi:                               0.10.2-py36hed927a0_0      --> 0.10.2-py27h4f12af3_0
    jinja2:                             2.9.6-py36h10aa3a0_1       --> 2.9.6-py27hb525989_1
    jpeg:                               9b-vc14h4d7706e_1          [vc14] --> 9b-vc9hb90d6ad_1           [vc9]
    jsonschema:                         2.6.0-py36h7636477_0       --> 2.6.0-py27haaf3834_0
    jupyter:                            1.0.0-py36h422fd7e_2       --> 1.0.0-py27h9d2e098_2
    jupyter_client:                     5.1.0-py36h9902a9a_0       --> 5.1.0-py27hced1786_0
    jupyter_console:                    5.2.0-py36h6d89b47_1       --> 5.2.0-py27h6ed736b_1
    jupyter_core:                       4.3.0-py36h511e818_0       --> 4.3.0-py27h7bec3d2_0
    jupyterlab:                         0.27.0-py36h34cc53b_2      --> 0.27.0-py27h573f66c_2
    jupyterlab_launcher:                0.4.0-py36h22c3ccf_0       --> 0.4.0-py27ha72e593_0
    lazy-object-proxy:                  1.3.1-py36hd1c21d2_0       --> 1.3.1-py27ha5c8080_0
    libiconv:                           1.15-vc14h29686d3_5        [vc14] --> 1.15-vc9h6e1f958_5         [vc9]
    libpng:                             1.6.32-vc14hce43e6c_2      [vc14] --> 1.6.32-vc9h4f2627b_2       [vc9]
    libssh2:                            1.8.0-vc14hcf584a9_2       [vc14] --> 1.8.0-vc9h457d253_2        [vc9]
    libtiff:                            4.0.8-vc14h04e2a1e_10      [vc14] --> 4.0.8-vc9hf4dca05_10       [vc9]
    libxml2:                            2.9.4-vc14h8fd0f11_5       [vc14] --> 2.9.4-vc9h17753f3_5        [vc9]
    libxslt:                            1.1.29-vc14hf85b8d4_5      [vc14] --> 1.1.29-vc9h15c42d4_5       [vc9]
    llvmlite:                           0.20.0-py36_0              --> 0.20.0-py27_0
    locket:                             0.2.0-py36hfed976d_1       --> 0.2.0-py27h1ca288a_1
    lockfile:                           0.12.2-py36h0468280_0      --> 0.12.2-py27hcad0c83_0
    lxml:                               3.8.0-py36h45350b2_0       --> 3.8.0-py27h61b39f8_0
    markupsafe:                         1.0-py36h0e26971_1         --> 1.0-py27h9d4480d_1
    matplotlib:                         2.0.2-py36h58ba717_1       --> 2.0.2-py27hd2e917d_1
    mccabe:                             0.6.1-py36hb41005a_1       --> 0.6.1-py27hde0bf6e_1
    menuinst:                           1.4.8-py36h870ab7d_0       --> 1.4.8-py27h66acc81_0
    mistune:                            0.7.4-py36h4874169_0       --> 0.7.4-py27h0f41fca_0
    mkl-service:                        1.1.2-py36h57e144c_4       --> 1.1.2-py27h3c6b6b0_4
    mpmath:                             0.19-py36he326802_2        --> 0.19-py27h2868657_2
    msgpack-python:                     0.4.8-py36h58b1e9d_0       --> 0.4.8-py27he0366a1_0
    multipledispatch:                   0.4.9-py36he44c36e_0       --> 0.4.9-py27h8ebb51e_0
    navigator-updater:                  0.1.0-py36h8a7b86b_0       --> 0.1.0-py27h5e42984_0
    nbconvert:                          5.3.1-py36h8dc0fde_0       --> 5.3.1-py27h7a573cf_0
    nbformat:                           4.4.0-py36h3a5bc1b_0       --> 4.4.0-py27hf49b375_0
    networkx:                           1.11-py36hdf4b0f5_0        --> 1.11-py27h5ef25a2_0
    nltk:                               3.2.4-py36hd0e0a39_0       --> 3.2.4-py27hfcc2c94_0
    nose:                               1.3.7-py36h1c3779e_2       --> 1.3.7-py27h84c72c6_2
    notebook:                           5.0.0-py36h27f7975_1       --> 5.0.0-py27h705c2fc_1
    numba:                              0.35.0-np113py36_10        --> 0.35.0-np113py27_10
    numexpr:                            2.6.2-py36h7ca04dc_1       --> 2.6.2-py27h3253787_1
    numpy:                              1.13.1-py36haf1bc54_2      --> 1.13.1-py27h0f1b411_2
    numpydoc:                           0.7.0-py36ha25429e_0       --> 0.7.0-py27hf7b062b_0
    odo:                                0.5.1-py36h7560279_0       --> 0.5.1-py27h64810b2_0
    olefile:                            0.44-py36h0a7bdd2_0        --> 0.44-py27h42675a9_0
    openpyxl:                           2.4.8-py36hf3b77f6_1       --> 2.4.8-py27hbb31d20_1
    openssl:                            1.0.2l-vc14hcac20b0_2      [vc14] --> 1.0.2l-vc9h02d068b_2       [vc9]
    packaging:                          16.8-py36ha0986f6_1        --> 16.8-py27hae1a450_1
    pandas:                             0.20.3-py36hce827b7_2      --> 0.20.3-py27he04484b_2
    pandocfilters:                      1.4.2-py36h3ef6317_1       --> 1.4.2-py27h76461d3_1
    partd:                              0.3.8-py36hc8e763b_0       --> 0.3.8-py27h1e0692f_0
    path.py:                            10.3.1-py36h3dd8b46_0      --> 10.3.1-py27hc415bcf_0
    pathlib2:                           2.3.0-py36h7bfb78b_0       --> 2.3.0-py27h0ae272f_0
    patsy:                              0.4.1-py36h42cefec_0       --> 0.4.1-py27h5f67e4e_0
    pep8:                               1.7.0-py36h0f3d67a_0       --> 1.7.0-py27h348567b_0
    pickleshare:                        0.7.4-py36h9de030f_0       --> 0.7.4-py27hb5f6335_0
    pillow:                             4.2.1-py36hdb25ab2_0       --> 4.2.1-py27hbae4915_0
    pip:                                9.0.1-py36hadba87b_3       --> 9.0.1-py27hdb66070_3
    pkginfo:                            1.4.1-py36hb0f9cfa_1       --> 1.4.1-py27h6ce81e3_1
    ply:                                3.10-py36h1211beb_0        --> 3.10-py27h5fb8a85_0
    progress:                           1.3-py36hbeca8d3_0         --> 1.3-py27hcb558ea_0
    prompt_toolkit:                     1.0.15-py36h60b8f86_0      --> 1.0.15-py27h3a8ec6a_0
    psutil:                             5.2.2-py36he84b5ac_0       --> 5.2.2-py27h4f0939b_0
    py:                                 1.4.34-py36ha4aca3a_1      --> 1.4.34-py27h50228bf_1
    pycodestyle:                        2.3.1-py36h7cc55cd_0       --> 2.3.1-py27h24cd5d9_0
    pycosat:                            0.6.2-py36hf17546d_1       --> 0.6.2-py27h96ae2ec_1
    pycparser:                          2.18-py36hd053e01_1        --> 2.18-py27hb43d16c_1
    pycrypto:                           2.6.1-py36he68e6e2_1       --> 2.6.1-py27h0cf9ff2_1
    pycurl:                             7.43.0-py36h086bf4c_3      --> 7.43.0-py27h4a6fa90_3
    pyflakes:                           1.5.0-py36h56100d8_1       --> 1.5.0-py27h610581c_1
    pygments:                           2.2.0-py36hb010967_0       --> 2.2.0-py27ha50f84f_0
    pylint:                             1.7.2-py36h091ff97_0       --> 1.7.2-py27h5ced823_0
    pyodbc:                             4.0.17-py36h0006bc2_0      --> 4.0.17-py27h79aec81_0
    pyopenssl:                          17.2.0-py36h15ca2fc_0      --> 17.2.0-py27h58ddd18_0
    pyparsing:                          2.2.0-py36h785a196_1       --> 2.2.0-py27hc7d9fa6_1
    pyqt:                               5.6.0-py36hb5ed885_5       --> 5.6.0-py27h224ed30_5
    pysocks:                            1.6.7-py36h698d350_1       --> 1.6.7-py27h59bdd1e_1
    pytables:                           3.4.2-py36h85b5e71_1       --> 3.4.2-py27h8d770d7_1
    pytest:                             3.2.1-py36h753b05e_1       --> 3.2.1-py27h7cb0baa_1
    python-dateutil:                    2.6.1-py36h509ddcb_1       --> 2.6.1-py27hbdcc174_1
    pytz:                               2017.2-py36h05d413f_1      --> 2017.2-py27haecbe8f_1
    pywavelets:                         0.5.2-py36hc649158_0       --> 0.5.2-py27h0dc3f48_0
    pywin32:                            221-py36h9c10281_0         --> 221-py27h5035e5b_0
    pyyaml:                             3.12-py36h1d1928f_1        --> 3.12-py27ha287073_1
    pyzmq:                              16.0.2-py36h38c27d9_2      --> 16.0.2-py27h7554bf6_2
    qt:                                 5.6.2-vc14h6f8c307_12      [vc14] --> 5.6.2-vc9hc26998b_12       [vc9]
    qtawesome:                          0.4.4-py36h5aa48f6_0       --> 0.4.4-py27h766b13d_0
    qtconsole:                          4.3.1-py36h99a29a9_0       --> 4.3.1-py27h77d40ac_0
    qtpy:                               1.3.1-py36hb8717c5_0       --> 1.3.1-py27h1ff2b4b_0
    requests:                           2.18.4-py36h4371aae_1      --> 2.18.4-py27h3159eba_1
    rope:                               0.10.5-py36hcaf5641_0      --> 0.10.5-py27h927ae9b_0
    ruamel_yaml:                        0.11.14-py36h9b16331_2     --> 0.11.14-py27hce2a281_2
    scikit-image:                       0.13.0-py36h6dffa3f_1      --> 0.13.0-py27h27e3c36_1
    scikit-learn:                       0.19.0-py36h294a771_2      --> 0.19.0-py27hc691070_2
    scipy:                              0.19.1-py36h7565378_3      --> 0.19.1-py27h069396b_3
    seaborn:                            0.8.0-py36h62cb67c_0       --> 0.8.0-py27h5729117_0
    setuptools:                         36.5.0-py36h65f9e6e_0      --> 36.5.0-py27ha483b79_0
    simplegeneric:                      0.8.1-py36heab741f_0       --> 0.8.1-py27h2c3a90d_0
    singledispatch:                     3.4.0.3-py36h17d0c80_0     --> 3.4.0.3-py27h3f9d112_0
    sip:                                4.18.1-py36h9c25514_2      --> 4.18.1-py27h5ec1c1a_2
    six:                                1.10.0-py36h2c0fdd8_1      --> 1.10.0-py27h5de879c_1
    snowballstemmer:                    1.2.1-py36h763602f_0       --> 1.2.1-py27h28d3bf7_0
    sortedcollections:                  0.5.3-py36hbefa0ab_0       --> 0.5.3-py27h21b938c_0
    sortedcontainers:                   1.5.7-py36ha90ac20_0       --> 1.5.7-py27hb302113_0
    sphinx:                             1.6.3-py36h9bb690b_0       --> 1.6.3-py27h858dd0f_0
    sphinxcontrib:                      1.0-py36hbbac3d2_1         --> 1.0-py27h0e2fb95_1
    sphinxcontrib-websupport:           1.0.1-py36hb5e5916_1       --> 1.0.1-py27h0d0f901_1
    spyder:                             3.2.3-py36h132a2c8_0       --> 3.2.3-py27hbf30467_0
    sqlalchemy:                         1.1.13-py36h5948d12_0      --> 1.1.13-py27had82547_0
    sqlite:                             3.20.1-vc14h7ce8c62_1      [vc14] --> 3.20.1-vc9h5ec81ce_1       [vc9]
    statsmodels:                        0.8.0-py36h6189b4c_0       --> 0.8.0-py27hc77c1fc_0
    sympy:                              1.1.1-py36h96708e0_0       --> 1.1.1-py27hde44fae_0
    tblib:                              1.3.2-py36h30f5020_0       --> 1.3.2-py27h8ae915c_0
    testpath:                           0.3.1-py36h2698cfe_0       --> 0.3.1-py27h1cd488d_0
    tk:                                 8.6.7-vc14hb68737d_1       [vc14] --> 8.6.7-vc9h452fa07_1        [vc9]
    toolz:                              0.8.2-py36he152a52_0       --> 0.8.2-py27h1873a47_0
    tornado:                            4.5.2-py36h57f6048_0       --> 4.5.2-py27hbdd278e_0
    traitlets:                          4.3.2-py36h096827d_0       --> 4.3.2-py27h1b1b3a5_0
    typing:                             3.6.2-py36hb035bda_0       --> 3.6.2-py27h9197bb0_0
    unicodecsv:                         0.14.1-py36h6450c06_0      --> 0.14.1-py27h0bf7bb0_0
    urllib3:                            1.22-py36h276f60a_0        --> 1.22-py27hb9f5a07_0
    wcwidth:                            0.1.7-py36h3d5aa90_0       --> 0.1.7-py27hb1a0d82_0
    webencodings:                       0.5.1-py36h67c50ae_1       --> 0.5.1-py27h4e224a2_1
    werkzeug:                           0.12.2-py36h866a736_0      --> 0.12.2-py27h822ccf9_0
    wheel:                              0.29.0-py36h6ce6cde_1      --> 0.29.0-py27hffb40b6_1
    widgetsnbextension:                 3.0.2-py36h364476f_1       --> 3.0.2-py27h599da02_1
    win_inet_pton:                      1.0.1-py36he67d7fd_1       --> 1.0.1-py27hf41312a_1
    win_unicode_console:                0.5-py36hcdbd4b5_0         --> 0.5-py27hc037021_0
    wincertstore:                       0.2-py36h7fe50ca_0         --> 0.2-py27hf04cefb_0
    wrapt:                              1.10.11-py36he5f5981_0     --> 1.10.11-py27hcd2b27d_0
    xlrd:                               1.1.0-py36h1cb58dc_1       --> 1.1.0-py27h2b87a7f_1
    xlsxwriter:                         0.9.8-py36hdf8fb07_0       --> 0.9.8-py27h3916130_0
    xlwings:                            0.11.4-py36hd3cf94d_0      --> 0.11.4-py27h0373cff_0
    xlwt:                               1.3.0-py36h1a4751e_0       --> 1.3.0-py27h2271735_0
    yaml:                               0.1.7-vc14hb31d195_1       [vc14] --> 0.1.7-vc9heb3852e_1        [vc9]
    zict:                               0.1.2-py36hbda90c0_0       --> 0.1.2-py27h59bcc41_0
    zlib:                               1.2.11-vc14h1cdd9ab_1      [vc14] --> 1.2.11-vc9ha80f4e8_1       [vc9]

The following packages will be DOWNGRADED:

    ipython:                            6.1.0-py36h236ecc8_1       --> 5.4.1-py27hbf971b4_1
    python:                             3.6.2-h6679aeb_11          --> 2.7.13-hb034564_12
    vc:                                 14-h2379b0c_1              --> 9-h7299396_0
```

**Comment by mcg1969** [comment]
conda 4.3.30, `conda install git --dry-run`:
```
Fetching package metadata .............
Solving package specifications: .
Warning: >10 possible package resolutions (only showing differing packages):
  - defaults::curl-7.52.1-vc14_0, defaults::freetype-2.5.5-vc14_2, defaults::hdf5-1.8.15.1-vc14_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc14_2, defaults::hdf5-1.8.15.1-vc14_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc14_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc14_0, defaults::freetype-2.5.5-vc14_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc14_0, defaults::freetype-2.5.5-vc9_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc14_0, defaults::freetype-2.5.5-vc9_2, defaults::hdf5-1.8.15.1-vc14_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc9_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc9_2, defaults::hdf5-1.8.15.1-vc14_4, defaults::tk-8.5.18-vc9_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc9_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc14_0
  - defaults::curl-7.52.1-vc9_0, defaults::freetype-2.5.5-vc14_2, defaults::hdf5-1.8.15.1-vc9_4, defaults::tk-8.5.18-vc14_0
  ... and others

Package plan for installation in environment C:\Users\mcg\Anaconda3:

The following NEW packages will be INSTALLED:

    git:               2.14.1-h8288424_2
    vs2008_runtime:    9.00.30729.1-hfaea7d5_0

The following packages will be UPDATED:

    anaconda:          5.0.0-py36hea9b2fc_0    --> custom-py36_0
    bitarray:          0.8.1-py36h6af124b_0    --> 0.8.1-py36_1
    bottleneck:        1.2.1-py36hd119dfa_0    --> 1.2.1-np113py36_0
    bzip2:             1.0.6-vc14hdec8e7a_1    [vc14] --> 1.0.6-vc9h563fc2e_1  [vc9 ]
    cytoolz:           0.8.2-py36h547e66e_0    --> 0.8.2-py36_0
    fastcache:         1.0.2-py36hffdae1b_0    --> 1.0.2-py36_1
    gevent:            1.2.2-py36h342a76c_0    --> 1.2.2-py36_0
    greenlet:          0.4.12-py36ha00ad21_0   --> 0.4.12-py36_0
    icu:               58.2-vc14hc45fdbb_0     [vc14] --> 58.2-vc9h3e79e4e_0   [vc9 ]
    jpeg:              9b-vc14h4d7706e_1       [vc14] --> 9b-vc9hb90d6ad_1     [vc9 ]
    lazy-object-proxy: 1.3.1-py36hd1c21d2_0    --> 1.3.1-py36_0
    libiconv:          1.15-vc14h29686d3_5     [vc14] --> 1.15-vc9h6e1f958_5   [vc9 ]
    libpng:            1.6.32-vc14hce43e6c_2   [vc14] --> 1.6.32-vc9h2352712_3 [vc9 ]
    libssh2:           1.8.0-vc14hcf584a9_2    [vc14] --> 1.8.0-vc9h457d253_2  [vc9 ]
    libtiff:           4.0.8-vc14h04e2a1e_10   [vc14] --> 4.0.8-vc9hf4dca05_10 [vc9 ]
    libxml2:           2.9.4-vc14h8fd0f11_5    [vc14] --> 2.9.4-vc9h17753f3_5  [vc9 ]
    libxslt:           1.1.29-vc14hf85b8d4_5   [vc14] --> 1.1.29-vc9h15c42d4_5 [vc9 ]
    lxml:              3.8.0-py36h45350b2_0    --> 3.8.0-py36_0
    msgpack-python:    0.4.8-py36h58b1e9d_0    --> 0.4.8-py36_0
    openssl:           1.0.2l-vc14hcac20b0_2   [vc14] --> 1.0.2l-vc9h02d068b_2 [vc9 ]
    pillow:            4.2.1-py36hdb25ab2_0    --> 4.2.1-py36_0
    psutil:            5.2.2-py36he84b5ac_0    --> 5.2.2-py36_0
    pycrypto:          2.6.1-py36he68e6e2_1    --> 2.6.1-py36_6
    pyodbc:            4.0.17-py36h0006bc2_0   --> 4.0.17-py36_0
    pywavelets:        0.5.2-py36hc649158_0    --> 0.5.2-np113py36_0
    qt:                5.6.2-vc14h6f8c307_12   [vc14] --> 5.6.2-vc9hc26998b_12 [vc9 ]
    sqlalchemy:        1.1.13-py36h5948d12_0   --> 1.1.13-py36_0
    sqlite:            3.20.1-vc14h7ce8c62_1   [vc14] --> 3.20.1-vc9h5ec81ce_1 [vc9 ]
    statsmodels:       0.8.0-py36h6189b4c_0    --> 0.8.0-np113py36_0
    wrapt:             1.10.11-py36he5f5981_0  --> 1.10.11-py36_0
    yaml:              0.1.7-vc14hb31d195_1    [vc14] --> 0.1.7-vc9heb3852e_1  [vc9 ]
    zlib:              1.2.11-vc14h1cdd9ab_1   [vc14] --> 1.2.11-vc9ha80f4e8_1 [vc9 ]

The following packages will be DOWNGRADED:

    astropy:           2.0.2-py36h06391c4_4    --> 2.0.1-np113py36_0
    cffi:              1.10.0-py36hae3d1b5_1   --> 1.10.0-py36_0
    cryptography:      2.0.3-py36h123decb_1    --> 1.8.1-py36_0
    curl:              7.55.1-vc14hdaba4a4_3   [vc14] --> 7.52.1-vc14_0        [vc14]
    cython:            0.26.1-py36h18049ac_0   --> 0.26-py36_0
    freetype:          2.8-vc14h17c9bdf_0      [vc14] --> 2.5.5-vc14_2         [vc14]
    h5py:              2.7.0-py36hfbe0a52_1    --> 2.7.0-np113py36_0
    hdf5:              1.10.1-vc14hb361328_0   [vc14] --> 1.8.15.1-vc14_4      [vc14]
    markupsafe:        1.0-py36h0e26971_1      --> 1.0-py36_0
    matplotlib:        2.0.2-py36h58ba717_1    --> 2.0.2-np113py36_0
    menuinst:          1.4.8-py36h870ab7d_0    --> 1.4.7-py36_0
    mkl:               2018.0.0-h36b65af_4     --> 2017.0.3-0
    mkl-service:       1.1.2-py36h57e144c_4    --> 1.1.2-py36_3
    numexpr:           2.6.2-py36h7ca04dc_1    --> 2.6.2-np113py36_0
    numpy:             1.13.1-py36haf1bc54_2   --> 1.13.1-py36_0
    pandas:            0.20.3-py36hce827b7_2   --> 0.20.3-py36_0
    pycosat:           0.6.2-py36hf17546d_1    --> 0.6.2-py36_0
    pycurl:            7.43.0-py36h086bf4c_3   --> 7.43.0-py36_2
    pyopenssl:         17.2.0-py36h15ca2fc_0   --> 17.0.0-py36_0
    pyqt:              5.6.0-py36hb5ed885_5    --> 5.6.0-py36_2
    pytables:          3.4.2-py36h85b5e71_1    --> 3.2.2-np113py36_4
    python:            3.6.2-h6679aeb_11       --> 3.6.2-0
    pywin32:           221-py36h9c10281_0      --> 220-py36_2
    pyyaml:            3.12-py36h1d1928f_1     --> 3.12-py36_0
    pyzmq:             16.0.2-py36h38c27d9_2   --> 16.0.2-py36_0
    ruamel_yaml:       0.11.14-py36h9b16331_2  --> 0.11.14-py36_1
    scikit-image:      0.13.0-py36h6dffa3f_1   --> 0.13.0-np113py36_0
    scikit-learn:      0.19.0-py36h294a771_2   --> 0.19.0-np113py36_0
    scipy:             0.19.1-py36h7565378_3   --> 0.19.1-np113py36_0
    sip:               4.18.1-py36h9c25514_2   --> 4.18-py36_0
    tk:                8.6.7-vc14hb68737d_1    [vc14] --> 8.5.18-vc9_0         [vc9 ]
    vc:                14-h2379b0c_1           --> 9-h7299396_0
```

**Comment by mcg1969** [comment]
conda 4.3.27, `conda install git python=3.6`. This leads me to believe that the issue I identified with `add_defaults_to_specs` and timestamps is at play.
```
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment C:\Users\mcg\Anaconda3:

The following NEW packages will be INSTALLED:

    git:      2.11.1-0

The following packages will be UPDATED:

    anaconda: 5.0.0-py36hea9b2fc_0  --> custom-py36_0
    conda:    4.3.27-py36hcbae3bd_0 --> 4.3.30-py36h7e176b0_0
    python:   3.6.2-h6679aeb_11     --> 3.6.3-h9e2ca53_1
```

**Comment by amueller** [comment]
yeah that looks like what I got, thanks for investigating.

**Comment by mcg1969** [comment]
conda 4.3.30, `conda install git python=3.6`:
```
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment C:\Users\mcg\Anaconda3:

The following NEW packages will be INSTALLED:

    git:      2.11.1-0

The following packages will be UPDATED:

    anaconda: 5.0.0-py36hea9b2fc_0 --> custom-py36_0
    python:   3.6.2-h6679aeb_11    --> 3.6.3-h9e2ca53_1
```

**Comment by mcg1969** [comment]
Note the `git` version differences here. I think that's the key.

**Comment by mcg1969** [comment]
Here's what I think is happening.

With `conda install git`: conda first maximizes the version, build, and timestamp of `git`, and obtains `2.14.1 h8288424_2`. This introduces a `vc 9.*` dependency, which in turn introduces the `vc9` feature. Conda then determines that it can do without `vc9` _and_ `vc14` by downgrading Python. 

I'd say the primary problem here is that `git` should not be depending on vc9. I am _sure_ I have said this, multiple times: a package should _never_ include a `track_feature` package as a _dependency_. In this _particular_ case, `git` should depend on `vs2008_runtime` directly. But packages that really _need_ the `vc9` feature should specify it as a _feature_, not a dependency.

**Comment by mcg1969** [comment]
I'm a little confused by why conda 4.3.30 is keeping a few `vc14` packages around.

**Comment by mcg1969** [comment]
Just did a quick scan of `main`. Ugh. I'm seeing a ton of packages that bring in `vc 9`, `vc 14`, etc as a dependency instead of as a `features`/`requires_feature`. This is bound to be a problem.

**Comment by mingwandroid** [comment]
Looking into `git` it seems we forgot to put a `[unix]` selector on https://github.com/AnacondaRecipes/git-feedstock/blob/master/recipe/meta.yaml#L26

**Comment by msarahan** [comment]
@mcg1969 ``vc 9`` != ``vc9`` - the space is a package with a version number, the lack of space is the feature.  The package actually has the track_feature in it.

**Comment by mcg1969** [comment]
Yes, I know this. It’s still not correct. Features should not be pulled in by dependency.

**Comment by mcg1969** [comment]
I actually think we can probably live with the dependency. The primary problem is that Python _also_ has the feature. That _defintely_ needs to change. And packages like git that don’t need the exclusivity should depend on the runtime directly.

**Comment by kalefranz** [comment]
@mingwandroid @jjhelmus @nehaljwani @msarahan Does anybody *disagree* with removing `vc*` from python package's `track_features` field?  If not, I guess this'll be one of the first uses of https://gist.github.com/jjhelmus/bf9489e3847d8e1737afca7d04b2f66d once we get it implemented.

**Comment by msarahan** [comment]
> Features should not be pulled in by dependency.

This is *exactly* what has been happening all along with python.  Or am I misunderstanding?

@kalefranz I completely agree.

**Comment by mcg1969** [comment]
> This is exactly what has been happening all along with python. Or am I misunderstanding?

Yes, it has been wrong all along, and I have pointed it out in other issues, I'm reasonably sure. I'll try to find them.

**Comment by mcg1969** [comment]
I've been thinking about this more and I think there is some good news here. As long as we do the following two things, we'll be fine:

- Remove the `vc*` features from python
- Replace `vc*` dependencies with direct dependency runtimes for any non-Python package, like git.

The realization is this: this effectively makes the `vc` features _redundant and useless_. But that's fine, because the only thing we _really_ need here is _exclusivity_. That is: we need to make sure that only one `vc*` dependency is employed by all Python extensions. The solution generated by conda 4.3.30 above, which has _both_ `vc14` and `vc9` features present, is caused by the fact that `python` has a track-feature when it shouldn't. Perform the first fix above, and that problem goes away.

**Comment by mcg1969** [comment]
https://github.com/conda/conda/issues/5301#issuecomment-302212550

**Comment by mcg1969** [comment]
https://github.com/ContinuumIO/anaconda-issues/issues/1760

**Comment by mcg1969** [comment]
Ah, I wondered about this, and it looks to be right. [Here](https://github.com/conda/conda/pull/4878) is a PR that attempts to fix one lingering issue with this: `conda update --all` will attempt to upgrade `vc=9` to `vc=14` against a user's wishes. Now, with `conda 4.4`, this might not be a problem, if `vc=xxx` was ever explicitly specified. I'm not sure this is easily fixable otherwise, but we can certainly consider ways to do so. This might actually be an argument for keeping the features in place to provide an extra measure of "pinning".

**Comment by msarahan** [comment]
> The realization is this: this effectively makes the vc features redundant and useless. But that's fine, because the only thing we really need here is exclusivity. That is: we need to make sure that only one vc* dependency is employed by all Python extensions. And that exclusivity is accomplished just fine by the vc* packages themselves---not by the features.

Wow, yes, that's a very nice realization!  I'm very excited about this development.  I think your idea to keep the features for now is good, but I'll look forward to the day when we can get away without them.

**Comment by mcg1969** [comment]
Yeah, if we can git rid of “old-style” features whenever we really don’t need them, that would be a good thing. I think the issue raised by #4878 is worth pondering, though, to see if we can figure out a good way to deal with it. In conda 4.4, it might be as simple as having some sort of “auto-pin” flag that makes sure the installed version of “vc” is not include in an “update --all” command.

**Comment by mcg1969** [comment]
Actually, @kalefranz what we need is the _opposite_ of your `aggressive_update` idea

**Comment by jjhelmus** [comment]
I agree that the `vcX` features are no longer needed.  The `vc` package with versions numbers insures exclusivity.  For example, `pyyaml` needs to pull in a compatible `yaml` package, both need to require either `vc 14.*` or `vc 9.*`.  

Using `vcX` features alone does not guarantee this exclusivity. The solver would prefer not to include multiple features, but I do not believe it forbidden from creating an environment with a `vc9` featured `pyyaml` with a `vc14` featured `yaml`.

One thing that the `vcX` features does provide that the `vc` package does not is identifiable package filenames, i.e. **yaml-0.1.7-vc14hb31d195_1.tar.bz2** and **yaml-0.1.7-vc9heb3852e_1.tar.bz2**.  Without the `vcX` feature these would not include the **vc9**/**vc14** string and would be difficult to differentiate.  This may be a good reason to retain the features, although perhaps another solution could be used to retain this identifiable of packages.  

I also agree that self contained packages like `git` or `curl` should only require the appropriate `vs20XX_runtime` package and not the `vc` package.  It is entirely possible to use a `git` package which requires `vs2015_runtime` in a environment with `vc 9*` Python packages.  One difficult point here is knowing if a package is self contained.  `curl` is self contained, but `pycurl` uses it as a library and therefore the `curl` package likely should depend on the `vc` package.

**Comment by msarahan** [comment]
we need to split up curl into libcurl (just the library, needs to be tied to a specific vc) and curl (just the executable, can depend on a single, standalone msvc runtime).  I'll try to do that sometime soon.

**Comment by mingwandroid** [comment]
Git is compiled with mingw-w64/gcc and msys2/gcc and therefore requires no vc features or packages.

**Comment by kalefranz** [comment]
I think this issue was resolved with a series of metadata hotfixes regarding vc features several months ago.
